egs/fuss/baseline/train.py

```diff
@@ -0,0 +1,192 @@
+import os
+import argparse
+import json
+
+import torch
+from torch import nn
+from torch.optim.lr_scheduler import ReduceLROnPlateau
+from torch.utils.data import DataLoader
+import pytorch_lightning as pl
+from pytorch_lightning.callbacks import ModelCheckpoint, EarlyStopping
+
+from asteroid.engine.system import System
+from asteroid.data import FUSSDataset
+from asteroid.losses import PITLossWrapper, PairwiseNegSDR
+
+from model import make_model_and_optimizer
+
+
+# Keys which are not in the conf.yml file can be added here.
+# In the hierarchical dictionary created when parsing, the key `key` can be
+# found at dic['main_args'][key]
+
+# By default train.py will use all available GPUs. The `id` option in run.sh
+# will limit the number of available GPUs for train.py .
+# This can be changed: `python train.py --gpus 0,1` will only train on 2 GPUs.
+parser = argparse.ArgumentParser()
+parser.add_argument('--exp_dir', default='exp/tmp',
+                    help='Full path to save best validation model')
+
+
+EPS = 1e-8
+class ClipedSingleSrcSNR(nn.Module):
+    def __init__(self, active_clip=-30, inactive_clip=-20):
+        super().__init__()
+        self.active_clip = active_clip
+        self.inactive_clip = inactive_clip
+
+    def forward(self, est, target, mix=None):
+        loss = target.new_zeros(target.shape[0],)
+        target_norm = torch.norm(target, p=2, dim=-1)
+        silent_mask = target_norm < EPS  # (B,)
+        active_loss = - self.active_snr(est, target)[~silent_mask]
+        loss[~silent_mask] = torch.clamp(active_loss, min=self.active_clip)
+        # loss /= (1 - silent_mask.float().mean()) + EPS
+        inactive_loss = - self.inactive_snr(est, mix)[silent_mask]
+        loss[silent_mask] = torch.clamp(inactive_loss, min=self.inactive_clip)
+        return loss
+
+    @staticmethod
+    def active_snr(est, target):
+        den = torch.norm(target - est, dim=-1, p=2) + EPS
```

Comment:
@mpariente  shouldn't we use plain mean square ? (Giovanni Pepe pointed this out to me)

Reply:
At which point exactly? 
Computes the SNR, what should we replace by mean square?

Reply:
Srry did not notice the 20 in the log ... is the same thing